### ISSUE TITLE
Exclude "System.Composition.*" assemblies from Diagnostic.nuspec for the analyzer + code fix template project 

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#2710]
`Diagnostic.nuspec` file for the analyzer + code fix template project packages the following files:

```
  <files>
    <file src="*.dll" target="tools\analyzers\" exclude="**\Microsoft.CodeAnalysis.*;**\System.Collections.Immutable.*;**\System.Reflection.Metadata.*" />
    <file src="tools\*.ps1" target="tools\" />
  </files>
```

The exclude list should be updated to include _"System.Composition.*"_ assemblies, otherwise installing the nuget package also includes these System assemblies as analyzers:

![analyzer_nuget_packaging](https://cloud.githubusercontent.com/assets/10605811/7615635/b74c91be-f953-11e4-85f4-753a4f39ac07.png)

**Comment by srivatsn** [comment]
Why are the composition dlls in the tools\analyzers directory? That itself seems like a bug.

**Comment by jmarolf** [comment]
By default VSIX and nuget packagers copy everything from the output directory.  The fix here is simple, just add the additional filtering logic.  Do we now show assemblies even if they have no analyzers? What if an assembly is a dependency only, do we show it in solution explorer?

**Comment by mavasani** [comment]
@jmarolf Yes, analyzer dependencies with no analyzers are also shown in the solution explorer, as the screenshot shows.
@tmeschter can probably confirm if this is the desired behavior...

**Comment by sharwell** [comment]
Two things here:

1. Rather than use a wildcard include, why not use `<file src="{actual assembly name}.dll"/>`? If the user adds dependencies that need to be included in the package, updating the **.nuspec** is straightforward.
2. The [Tvl.NuGet.BuildTasks](https://www.nuget.org/packages/Tvl.NuGet.BuildTasks/) package allows you to add items to a project with the `NuGetManifest` item type, and it will package them during the build. You can see how I applied this to the output of a template here: DotNetAnalyzers/AsyncUsageAnalyzers@2b9a4e03fc43cf3a3317d15d631569137741b475 (also included are metadata updates to the NuGet specification, but it should still be clear what's going on. Unlike the post build action, this strategy is fully integrated with cleaning the project, is tracked as a project output, and can report errors during the build.

**Comment by sharwell** [comment]
This is the MSBuild output for "Minimal" when you are using the build tasks described in my previous comment for the AsyncUsageAnalyzers project:

```
1>------ Rebuild All started: Project: AsyncUsageAnalyzers, Configuration: Debug Any CPU ------
1>  AsyncUsageAnalyzers -> C:\dev\AsyncUsageAnalyzers\AsyncUsageAnalyzers\AsyncUsageAnalyzers\bin\Debug\AsyncUsageAnalyzers.dll
1>  AsyncUsageAnalyzers -> C:\dev\AsyncUsageAnalyzers\AsyncUsageAnalyzers\AsyncUsageAnalyzers\bin\Debug\AsyncUsageAnalyzers.1.0.0-dev.nupkg
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========
```

**Comment by mavasani** [comment]
@sharwell This particular step: _"If the user adds dependencies that need to be included in the package, updating the .nuspec is straightforward."_ might not be obvious to all analyzer authors, and the end user would end up running against analyzer load failures at runtime. Current behavior is more conservative and hence probably more desirable, but I'd let others comment on their preference.

**Comment by jmarolf** [comment]
fixed with internal Changeset 1471973
